sklearn/metrics/pairwise.py

```diff
@@ -271,38 +273,61 @@ def euclidean_distances(X, Y=None, *, Y_norm_squared=None, squared=False,
     """
     X, Y = check_pairwise_arrays(X, Y)
 
-    # If norms are passed as float32, they are unused. If arrays are passed as
-    # float32, norms needs to be recomputed on upcast chunks.
-    # TODO: use a float64 accumulator in row_norms to avoid the latter.
     if X_norm_squared is not None:
-        XX = check_array(X_norm_squared)
-        if XX.shape == (1, X.shape[0]):
-            XX = XX.T
-        elif XX.shape != (X.shape[0], 1):
+        X_norm_squared = check_array(X_norm_squared, ensure_2d=False)
+        if X_norm_squared.shape == (X.shape[0],):
+            X_norm_squared = X_norm_squared.reshape(-1, 1)
+        if X_norm_squared.shape == (1, X.shape[0]):
+            X_norm_squared = X_norm_squared.T
+        if X_norm_squared.shape != (X.shape[0], 1):
             raise ValueError(
                 "Incompatible dimensions for X and X_norm_squared")
```

Comment:
Please store the original shape of `X_norm_squared` (after the `check_array(X_norm_squared, ensure_2d=False)`) in a temp variable and report it in the `ValueError` message.